test/app/App/Notify/AddChain/index.test.js

```diff
@@ -0,0 +1,121 @@
+
+    const chainNameInput = getByLabelText('Chain Name')
+    expect(chainNameInput.value).toEqual('Chain Name')
+
+    const chainIdInput = getByLabelText('Chain ID')
+    expect(chainIdInput.value).toEqual('Chain ID')
+
+    const blockExplorerInput = getByLabelText('Block Explorer')
+    expect(blockExplorerInput.value).toEqual('Block Explorer')
+
+    const primaryRpcInput = getByLabelText('Primary RPC')
+    expect(primaryRpcInput.value).toEqual('Primary Endpoint')
+
+    const secondaryRpcInput = getByLabelText('Secondary RPC')
+    expect(secondaryRpcInput.value).toEqual('Secondary Endpoint')
+
+    const otherLayerButton = getByRole('radio', { name: 'Other' })
+    expect(otherLayerButton.getAttribute('aria-checked')).toBe('true')
+
+    const submitButton = getByRole('button')
+    expect(submitButton.textContent).toBe('Fill in Chain')
+  })
+
+  it('renders with the chain name from an add chain request', () => {
+    const props = requestProps({ name: 'Polygon' })
+    const { getByLabelText } = render(<AddChain {...props} />)
+
+    const chainNameInput = getByLabelText('Chain Name')
+    expect(chainNameInput.value).toEqual('Polygon')
+  })
+
+  it('renders with the chain ID from an add chain request', () => {
+    const props = requestProps({ id: '0x89' })
+    const { getByLabelText } = render(<AddChain {...props} />)
+
+    const chainIdInput = getByLabelText('Chain ID')
+    expect(chainIdInput.value).toEqual('137')
+  })
+
+  it('renders with the first block explorer from an add chain request', () => {
+    const props = requestProps({ blockExplorerUrls: ['https://polygonscan.com'] })
+    const { getByLabelText } = render(<AddChain {...props} />)
+
+    const blockExplorerInput = getByLabelText('Block Explorer')
+    expect(blockExplorerInput.value).toEqual('https://polygonscan.com')
+  })
+
+  it('renders with the first RPC url from an add chain request', () => {
+    const props = requestProps({ rpcUrls: ['https://myrpc.polygon.net'] })
+    const { getByLabelText } = render(<AddChain {...props} />)
+
+    const primaryRpcInput = getByLabelText('Primary RPC')
+    expect(primaryRpcInput.value).toEqual('https://myrpc.polygon.net')
+  })
+
+  it('renders with the second RPC url from an add chain request', () => {
+    const props = requestProps({ rpcUrls: ['https://disconnected.com', 'https://myrpc.polygon.net'] })
+    const { getByLabelText } = render(<AddChain {...props} />)
+
+    const secondaryRpcInput = getByLabelText('Secondary RPC')
+    expect(secondaryRpcInput.value).toEqual('https://myrpc.polygon.net')
+  })
+
+  it('allows a chain with valid settings to be created', () => {
+    const props = requestProps({ id: '0x89', name: 'Polygon' })
+    const { getByRole } = render(<AddChain {...props} />)
+
+    const submitButton = getByRole('button')
+    expect(submitButton.textContent).toBe('Add Chain')
+  })
+
+  it('does not allow a chain with an existing ID to be created', async () => {
```

Comment:
is this the only unhappy path here or do we need another test for e.g. missing input values?

Reply:
there are a lot of unhappy paths but I wasn't going to make the tests exhaustive. will add a couple more as make sense

Reply:
also this is mostly tested by the first test where none of the values are filled in